src/game/frontend/submenus/World/ObjectSpawner.cpp

```diff
@@ -0,0 +1,172 @@
+#include "core/commands/Command.hpp"
+#include "game/backend/FiberPool.hpp"
+#include "game/backend/NativeHooks.hpp"
+#include "game/backend/ScriptMgr.hpp"
+#include "game/backend/Self.hpp"
+#include "game/frontend/items/Items.hpp"
+#include "game/rdr/Natives.hpp"
+#include "game/rdr/data/ObjModels.hpp"
+#include "ObjectSpawner.hpp"
+#include "util/SpawnObject.hpp"
+
+namespace YimMenu::Submenus
+{
+	static bool IsObjectModelInList(const std::string& model)
+	{
+		for (const auto& objModel : YimMenu::Data::g_ObjModels)
+		{
+			if (objModel.model == model)
+				return true;
+		}
+		return false;
+	}
+
+	static int ObjectSpawnerInputCallback(ImGuiInputTextCallbackData* data)
+	{
+		if (data->EventFlag == ImGuiInputTextFlags_CallbackCompletion)
+		{
+			std::string newText;
+			std::string inputLower = data->Buf;
+			std::transform(inputLower.begin(), inputLower.end(), inputLower.begin(), ::tolower);
+
+			for (const auto& objModel : YimMenu::Data::g_ObjModels)
+			{
+				std::string modelLower = objModel.model;
+				std::transform(modelLower.begin(), modelLower.end(), modelLower.begin(), ::tolower);
+				if (modelLower.find(inputLower) != std::string::npos)
+				{
+					newText = objModel.model;
+					break;
+				}
+			}
+
+			if (!newText.empty())
+			{
+				data->DeleteChars(0, data->BufTextLen);
+				data->InsertChars(0, newText.c_str());
+			}
+
+			return 1;
+		}
+		return 0;
+	}
+
+	void RenderObjectSpawnerMenu()
+	{
+		static std::string objectModelBuffer;
+		static float positionOffsetX = 5.0f;
+		static float positionOffsetY = 5.0f;
+		static float positionOffsetZ = 0.0f;
+		static float pitch           = 0.0f;
+		static float yaw             = 0.0f;
+		static float roll            = 0.0f;
+		static float alpha           = 125.0f;
+		static bool onGround         = false;
+		static bool isFrozen         = false;
+		static bool isBurning        = false;
+		static bool isPickup         = false;
+		static bool showPreview      = false;
+		static bool hasCollision     = false;
+		static int modelHash         = 0;
+
+		Vector3 playerCoords  = Self::GetPed().GetPosition();
+		Vector3 forwardVector = ENTITY::GET_ENTITY_FORWARD_VECTOR(Self::GetPed().GetHandle());
+
+		Vector3 spawnPosition;
+		spawnPosition.x = playerCoords.x + forwardVector.x * positionOffsetX;
+		spawnPosition.y = playerCoords.y + forwardVector.y * positionOffsetY;
+		spawnPosition.z = playerCoords.z + positionOffsetZ;
+
+		char buffer[256];
+		strncpy(buffer, objectModelBuffer.c_str(), sizeof(buffer));
+
+		ImGui::InputTextWithHint("##objectmodel", "Object Model", buffer, sizeof(buffer), ImGuiInputTextFlags_CallbackCompletion, ObjectSpawnerInputCallback);
```

Comment:
You could use the frontend item `InputTextWithHint`. Make sure to call `.Draw()` if you choose to go this route, you probably should though.

Reply:
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	C2440	'<function-style-cast>': cannot convert from 'initializer list' to 'YimMenu::InputTextWithHint'

InputTextWithHint("##objectmodel", "Object Model", &buffer, ImGuiInputTextFlags_CallbackCompletion, nullptr, ObjectSpawnerInputCallback)
    .Draw();

Reply:
Buffer has to be a string, not a char.